pkg/grpc/api_test.go

```diff
@@ -1583,68 +1583,60 @@ func TestSubscribeConfiguration(t *testing.T) {
 	}
 
 	for _, tt := range testCases {
-		// Testing alpha1 endpoint
-		t.Run(tt.testName+"-alpha1", func(t *testing.T) {
-			req := &runtimev1pb.SubscribeConfigurationRequest{
-				StoreName: tt.storeName,
-				Keys:      tt.keys,
-			}
-
-			resp, _ := client.SubscribeConfigurationAlpha1(context.Background(), req)
-			if !tt.errorExcepted {
-				rsp, err := resp.Recv()
-				assert.NoError(t, err, "Expected no error")
-				assert.Equal(t, rsp.Items, tt.expectedResponse, "Expected response items to be same")
-			} else {
-				retry := 3
-				count := 0
-				_, err := resp.Recv()
-				for {
-					if err != nil {
-						break
-					}
-					if count > retry {
-						break
-					}
-					count++
-					time.Sleep(time.Millisecond * 10)
-					_, err = resp.Recv()
+		testFn := func(subscribeFn subscribeConfigurationFn) func(t *testing.T) {
+			return func(t *testing.T) {
+				req := &runtimev1pb.SubscribeConfigurationRequest{
+					StoreName: tt.storeName,
+					Keys:      tt.keys,
 				}
-				assert.Equal(t, tt.expectedError, status.Code(err))
-				assert.Error(t, err, "Expected error")
-			}
-		})
-		// Testing stable endpoint
-		t.Run(tt.testName, func(t *testing.T) {
-			req := &runtimev1pb.SubscribeConfigurationRequest{
-				StoreName: tt.storeName,
-				Keys:      tt.keys,
-			}
 
-			resp, _ := client.SubscribeConfiguration(context.Background(), req)
-			if !tt.errorExcepted {
-				rsp, err := resp.Recv()
-				assert.NoError(t, err, "Expected no error")
-				assert.Equal(t, rsp.Items, tt.expectedResponse, "Expected response items to be same")
-			} else {
-				retry := 3
-				count := 0
-				_, err := resp.Recv()
-				for {
-					if err != nil {
-						break
-					}
-					if count > retry {
-						break
+				resp, _ := subscribeFn(context.Background(), req)
```

Comment:
Checking the error here seems important.

Reply:
Agree and I'm tracking that in #6557